### PR TITLE
Fix `formatDocPage()` rendering blank rows for degenerate or hidden entry terms

### PR DESCRIPTION
**Pull request by dahlia**
## Summary

`formatDocPage()` did not guard against degenerate or hidden usage terms inside `DocEntry.term`, causing empty term shapes to render as blank rows or malformed placeholders in help output. This patch fixes two root causes and adds a new `context` option to `formatUsageTerm()`.

### Wrong hidden visibility predicate

`formatDocPage()` called `formatUsageTerm()`, which internally uses `filterUsageForDisplay()` to strip hidden terms. However, `filterUsageForDisplay()` always applied `isUsageHidden()`, which filters `hidden: true`, `"usage"`, and `"help"`. In the documentation context, `isDocHidden()` should be used instead, which filters `hidden: true`, `"doc"`, and `"help"`. As a result, terms with `hidden: "doc"` leaked into doc output, while terms with `hidden: "usage"` were incorrectly omitted.

To fix this, `filterUsageForDisplay()`, `formatUsageTerms()`, and `formatUsageTermInternal()` (all internal functions in *usage.ts*) now accept an `isHidden` predicate parameter. A new `context` option on `UsageTermFormatOptions` lets callers select the appropriate filtering: `"usage"` (default) or `"doc"`. `formatDocPage()` now passes `context: "doc"`.

### No blank-row guard for degenerate terms

Even after hidden filtering, structurally degenerate terms could still produce blank or malformed output:

```typescript
// Option with no names renders as a bare metavar with a leading space
{ type: "option", names: [], metavar: "X" }

// Empty command/argument/literal renders as a blank row
{ type: "command", name: "" }
{ type: "argument", metavar: "" }
{ type: "literal", value: "" }

// Empty exclusive renders as "()" through formatUsageTermInternal
{ type: "exclusive", terms: [] }
```

`formatUsageTermInternal()` now returns early for options with `names.length === 0` (which also prevents a `reduce()` crash when `onlyShortestOptions` is set). Additionally, `formatDocPage()` now pre-filters all section entries before validation and rendering: each entry's term is rendered with `formatUsageTerm({ context: "doc" })` and entries whose result is visually blank are removed. This pre-filtering step in *doc.ts* ensures that:

- Titled sections whose entries are all filtered do not emit dangling section headers.
- `maxWidth` validation only considers entries that will actually render, so pages whose only entries are hidden no longer throw `RangeError` for small `maxWidth` values.

Closes https://github.com/dahlia/optique/issues/488

## Test plan

- [x] `formatUsageTerm()` with `context: "doc"` correctly hides `hidden: "doc"` and `hidden: "help"` terms, shows `hidden: "usage"` terms, and handles nested hidden terms in exclusive groups
- [x] `formatUsageTerm()` returns empty string for option with `names: []`
- [x] `formatDocPage()` skips entries with `hidden: true`, `hidden: "doc"`, degenerate option/command/argument/literal/exclusive terms
- [x] `formatDocPage()` still shows entries with `hidden: "usage"` terms
- [x] Section headings are suppressed when all entries in a section are filtered out
- [x] `maxWidth` validation does not reject pages where all entries are filtered
- [x] All existing tests pass across Deno, Node.js, and Bun (`mise test`)

### CODE REVIEW

File: packages/core/src/usage.test.ts [hunk 1]
```diff
@@ -1629,6 +1629,90 @@ describe("formatUsageTerm", () => {
...
+      };
+      const result = formatUsageTerm(term, { context: "doc" });
+      assert.equal(result, "(--quiet)");
+    });
+
+    it("should return empty for option with empty names", () => {
+      const term: UsageTerm = {
+        type: "option",
+        names: [] as unknown as readonly OptionName[],
+        metavar: "X",
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/usage.test.ts:1748]: Fixed in c4537c4.

File: packages/core/src/usage.ts [hunk 2]
```diff
@@ -636,6 +640,17 @@ export interface UsageTermFormatOptions extends UsageFormatOptions {
...
+  /**
+   * The rendering context, which determines which hidden visibility values
+   * cause terms to be filtered out.
+   *
+   * - `"usage"` (default): filters terms hidden from usage output
+   * - `"doc"`: filters terms hidden from documentation output
+   * @default `"usage"`
+   * @since 0.11.0
+   */
+  readonly context?: "usage" | "doc";
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/usage.ts:649]: Fixed in c4537c4.

**Comment by dahlia** [comment]
@coderabbitai full review

File: packages/core/src/usage.ts [hunk 3]
```diff
@@ -608,21 +615,22 @@ function filterUsageForDisplay(usage: Usage): Usage {
...
   for (const t of terms) {
     if (
       "hidden" in t &&
       (t.type === "argument" || t.type === "option" || t.type === "command" ||
         t.type === "passthrough") &&
-      isUsageHidden(t.hidden)
+      isHidden(t.hidden)
     ) {
       continue;
     }
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/usage.ts]: Fixed in 6d3041c. Removed the redundant `isHidden` check from `formatUsageTerms()` along with the `isHidden` parameter from both `formatUsageTerms()` and `formatUsageTermInternal()`. All hidden filtering is now solely in `filterUsageForDisplay()`.

File: packages/core/src/usage.ts [hunk 4]
```diff
@@ -558,25 +558,32 @@ function normalizeUsageTerm(term: UsageTerm): UsageTerm {
...
         term.type === "command" || term.type === "passthrough") &&
-      isUsageHidden(term.hidden)
+      isHidden(term.hidden)
     ) {
       continue;
     }
+    // Skip degenerate zero-width leaf terms
+    if (term.type === "option" && term.names.length === 0) continue;
+    if (term.type === "command" && term.name === "") continue;
+    if (term.type === "literal" && term.value === "") continue;
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/usage.ts:578]: Fixed in 568e530. Added `term.metavar.length === 0` check (using `.length` instead of `=== ""` to avoid a TS2367 error since `metavar` is typed as `NonEmptyString`).

File: packages/core/src/usage.ts [hunk 5]
```diff
@@ -683,6 +695,7 @@ function* formatUsageTermInternal(
       width: term.metavar.length,
     };
   } else if (term.type === "option") {
+    if (term.names.length < 1) return;
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/usage.ts]: Fixed in 8976859. Removed the redundant guard since filterUsageForDisplay() already filters out option terms with empty names.

File: packages/core/src/doc.test.ts [hunk 6]
```diff
@@ -2323,4 +2323,174 @@ describe("branch coverage: doc.ts edge cases", () => {
...
+              hidden: "usage",
+            } as never,
+            description: [{ type: "text", text: "Verbose output" }],
+          }],
+        }],
+      };
+      const result = formatDocPage("app", page, { colors: false });
+      assert.ok(result.includes("--verbose"));
+      assert.ok(result.includes("Verbose output"));
+    });
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/doc.test.ts:2400]: Fixed in fd4d479. Added a test for `hidden: "help"` in the `formatDocPage` suite.

File: packages/core/src/doc.test.ts [hunk 7]
```diff
@@ -2323,4 +2323,174 @@ describe("branch coverage: doc.ts edge cases", () => {
...
+        sections: [{
+          entries: [{
+            term: { type: "exclusive", terms: [] } as never,
+          }],
+        }],
+      };
+      const result = formatDocPage("app", page, { colors: false });
+      const lines = result.split("\n").filter((l) => l.trim() !== "");
+      assert.equal(lines.length, 0);
+    });
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/doc.test.ts:2470]: Skipping this one — the pattern only repeats a few times and the project guidelines prefer avoiding premature abstractions for small repetitions.

**Comment by dahlia** [comment]
@coderabbitai full review
